File: cms-app-crawler-ts/src/utils/JsonDataProvider.ts
import fs from 'fs';
import path from 'path';

const TESTDATA_DIR = path.join(__dirname, '../../resources/testdata');

function getData<T = Record<string, string>>(methodName: string): T[] {
  const filePath = path.join(TESTDATA_DIR, `${methodName}.json`);
  if (!fs.existsSync(filePath)) {
    throw new Error(`No test data JSON found for: ${methodName} (expected: ${filePath})`);
  }
  const rows = JSON.parse(fs.readFileSync(filePath, 'utf8'));
  if (!Array.isArray(rows)) {
    throw new Error(`Test data file must contain a JSON array: ${filePath}`);
  }
  return rows as T[];
}

export const JsonDataProvider = { getData };

File: cms-app-crawler-py/src/utils/json_data_provider.py
import json
from pathlib import Path

_TESTDATA_DIR = Path(__file__).parent.parent.parent / "resources" / "testdata"


class JsonDataProvider:
    @staticmethod
    def get_data(method_name: str) -> list[dict]:
        file_path = _TESTDATA_DIR / f"{method_name}.json"
        if not file_path.exists():
            raise FileNotFoundError(
                f"No test data JSON found for: {method_name} (expected: {file_path})"
            )
        with open(file_path, encoding="utf-8") as f:
            rows = json.load(f)
        if not isinstance(rows, list):
            raise ValueError(f"Test data file must contain a JSON array: {file_path}")
        return rows
